### PR TITLE
Move text-font property from PointAnnotation to PointAnnotationManager

### PR DESCRIPTION
**Pull request by Chaoba**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).
Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

## Pull request checklist:
 - [ ] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [x] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [x] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog>Move text-font property from PointAnnotation to PointAnnotationManager</changelog>`.

### Summary of changes
Property `text-font` is a data-driven property, however, if we apply this property in `PointAnnotationOptions`

```
  val pointAnnotationOptions: PointAnnotationOptions = PointAnnotationOptions()
     .withTextFont( listOf( "Arial Unicode MS Bold",   "Open Sans Regular" ))
```
We will have the following error:
```
Layer 'mapbox-android-symbol-layer-4' has an invalid value for text-font and will not render text. Output values must be contained as literals within the expression.
``` 
This is because we are using `get` expression to apply data-driven properties with outputs not predictable.
Engine must know what are the possible outputs of an expression. Example:

- Invalid case: `"text-font": [ "get", "text-font" ]`, evaluation result can be anything, no way to know what is the output of that expression
- Valid case: `"text-font": ["match", ["get", "text-font"], "iOS", ["literal", ["SanFrancisco"]], ["literal", ["Roboto"]]]`, there are only two possible evaluation outputs: "SanFrancisco" or "Roboto"

This pr makes `text-font` not data-driven anymore.

### User impact (optional)
Need to change text font config from `PointAnnotationOptions`
```
  val pointAnnotationOptions: PointAnnotationOptions = PointAnnotationOptions()
     .withTextFont( listOf( "Arial Unicode MS Bold",   "Open Sans Regular" ))
```
to `PointAnnotationManager`
```
 symbolManager = annotationPlugin.createPointAnnotationManager(mapView).apply {textFont = listOf("Arial Unicode MS Bold", "Open Sans Regular")
```

### CODE REVIEW

File: plugin-annotation/src/main/java/com/mapbox/maps/plugin/annotation/generated/PointAnnotationManager.kt [hunk 1]
```diff
@@ -111,7 +109,7 @@ class PointAnnotationManager(
...
...tAnnotationOptions.PROPERTY_TEXT_JUSTIFY - TextJustify   * PointAnnotationOptions.PROPERTY_TEXT_LETTER_SPACING - Double   * PointAnnotationOptions.PROPERTY_TEXT_MAX_WIDTH - Double   * PointAnnotationOptions.PROPERTY_TEXT_OFFSET - List<Double>   * PointAnnotationOptions.PROPERTY_TEXT_RADIAL_OFFSET - Double   * PointAnnotationOptions.PROPERTY_TEXT_ROTATE - Double   * PointAnnotationOptions.PROPERTY_TEXT_SIZE - Double   * PointAnnotationOptions.PROPERTY_TEXT_TRANSFORM - TextTransform   * PointAnnotationOptions.PROPERTY_ICON_COLOR - String   * PointAnnotationOptions.PROPERTY_ICON_HALO_BLUR - Double   * PointAnnotationOptions.PROPERTY_ICON_HALO_COLOR - String   * PointAnnotationOptions.PROPERTY_ICON_HALO_WIDTH - Double   * PointAnnotationOptions.PROPERTY_ICON_OPACITY - Double   * PointAnnotationOptions.PROPERTY_TEXT_COLOR - String   * PointAnnotationOptions.PROPERTY_TEXT_HALO_BLUR - Double   * PointAnnotationOptions.PROPERTY_TEXT_HALO_COLOR - String   * PointAnnotationOptions.PROPERTY_TEXT_HALO_WIDTH - Double   * PointAnnotationOptions.PROPERTY_TEXT_OPACITY - Double   * Learn more about above properties in the )[The online documentation](https://www.mapbox.com/mapbox-gl-js/style-spec/).
```
Comment (pengdev) [inline plugin-annotation/src/main/java/com/mapbox/maps/plugin/annotation/generated/PointAnnotationManager.kt:112]: nit, can we add a new line after each property?

File: plugin-annotation/src/main/java/com/mapbox/maps/plugin/annotation/generated/PointAnnotationManager.kt [hunk 2]
```diff
@@ -469,6 +519,30 @@ class PointAnnotationManager(
       }
     }
 
+  /**
+   * The TextFont property
+   *
+   * Font stack to use for displaying text.
+   */
+  var textFont: List<String>?
```
Comment (kiryldz) [inline plugin-annotation/src/main/java/com/mapbox/maps/plugin/annotation/generated/PointAnnotationManager.kt:527]: If it's null, what default value for text font?

**Review by kiryldz — APPROVED** [review]
Looks good, please revisit PR description, I see several typos there and root cause issue is not that clear.
Also if it's breaking change - please add section `User impact` showing how to properly apply this breaking change.

File: plugin-annotation/src/main/java/com/mapbox/maps/plugin/annotation/generated/PointAnnotationManager.kt (hunk 2 above)
Comment (pengdev) [reply plugin-annotation/src/main/java/com/mapbox/maps/plugin/annotation/generated/PointAnnotationManager.kt:527]: shall we include the default value here?(maybe it's not required though)

**Pull request merged**